main.go

```diff
@@ -0,0 +1,475 @@
+	allProjects, err := findResourcesByAssetType(ctx, *organization, assetTypeProject, emptyQuery, assetClient)
+	if err != nil {
+		glog.Errorf("Error finding all projects in the organization %v", err)
+		return
+	}
+
+	glog.V(20).Infoln("      Getting Users")
+
+	// Initialize the Workspace Admin client
+	// This client will be used to find users in a given Cloud Identity/Workspace domain
+	adminService, err := getAdminServiceClient(ctx, impersonateAccount, svcKeyCred, *subject)
+	if err != nil {
+		glog.Errorf("Error initializing admin client %v", err)
+		return
+	}
+
+	// If you want to narrow the search to a subset of users, apply a searchFilter here
+	//  https://developers.google.com/admin-sdk/directory/v1/guides/search-users
+	//	eg. searchFilter := "isAdmin=false"
+
+	searchFilter := ""
+	allUsers, err := findDomainUsers(ctx, *cx, searchFilter, adminService)
+	if err != nil {
+		glog.Errorf("Error finding domain users %v", err)
+		return
+	}
+
+	// Launch a goroutines that will process the projects and organizations
+	// that are visible to users.  The channel is fed by the
+	// getOrganizations(), and getProjects() function calls in the succeeding stanza.
+	ch := make(chan *userAccess)
+	done := make(chan bool)
+	go func() {
+		for {
+			msg, ok := <-ch
+			if ok {
+				if msg.Error != nil {
+					glog.Errorf("Error iterating: %v", msg.Error)
+				} else {
+					for _, o := range msg.Organizations {
+						glog.V(50).Infof("             User %s has Organization visibility to %s", msg.User.PrimaryEmail, o.Name)
+						if _, ok := allOrganizations[o.Name]; !ok {
+							glog.V(2).Infof("             User [%s] has external organization visibility to [%s](%s)", msg.User.PrimaryEmail, o.Name, o.DisplayName)
+						}
+					}
+					for _, p := range msg.Projects {
+						glog.V(50).Infof("             User %s has Project visibility to %s", msg.User.PrimaryEmail, p.ProjectId)
+						if _, ok := allProjects[p.ProjectId]; !ok {
+							glog.V(2).Infof("             User [%s] has external project visibility to [projects/%d](%s)", msg.User.PrimaryEmail, p.ProjectNumber, p.ProjectId)
+						}
+					}
+				}
+			} else {
+				done <- true
+				return
+			}
+		}
+	}()
+
+	// Launch goroutines for each user that feeds a channel received the the go routine defined in the preceding stanza.
+	for _, u := range allUsers {
+
+		// Select a random service account to use per user and disribute quota consumption
+		impersonateAccount, svcKeyCred := getRandomServiceAccount(serviceAccounts, svcAccountKeys)
+
+		// We need to iterate over all organizations visible to the current user
+		// The noFilter is explicitly defined here to allow you to modify which organizations to
+		// limit the query.
+		// see https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v1#SearchOrganizationsRequest
+		noFilter := ""
+		wg.Add(1)
+		go getOrganizations(ctx, ch, &wg, limiter, noFilter, impersonateAccount, svcKeyCred, *u)
```

Comment:
Consider doing something like this:

```go
go func() {
    wg.Add(1)
    defer wg.Done()
    userOrgs, err := getOrganizations(ctx, ...)
    if err != nil { ... }
    userProjects := getProjects(ctx, ...)
    if err != nil { ... }
    ch <- userAccess{...} // No need for the Error field anymore.
}()
```

The organizations and projects calls don't happen concurrently, but that's OK because there is enough else going on concurrently it shouldn't matter in the end. Benefits:

* You don't have to pass the channel to `getOrganization` or `getProjects` -- they don't need to know they're being called concurrently at all; they just return the values they were asked for.
* You also don't have to pass `wg`.
* The organization and project info for each user is all printed next to each other, not whatever order the runtime scheduler decides.